contrib/internal/httputil/trace_test.go

```diff
@@ -182,3 +182,38 @@ func TestTraceAndServe(t *testing.T) {
 		assert.Equal("200", spans[0].Tag(ext.HTTPCode))
 	})
 }
+
+func TestTraceAndServeHost(t *testing.T) {
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}
+
+	t.Run("on", func(t *testing.T) {
+		mt := mocktracer.Start()
+		assert := assert.New(t)
+		defer mt.Stop()
+
+		w := httptest.NewRecorder()
+		r, err := http.NewRequest("GET", "http://localhost/", nil)
+		assert.NoError(err)
+		TraceAndServe(http.HandlerFunc(handler), w, r, "service", "resource", nil)
```

Comment:
Can we remove the `w` variable in both tests and just do it inline? As a reader I tend to think it might be used elsewhere too...
```suggestion
		TraceAndServe(handler, http.NewRecorder(), r, "service", "resource", nil)
```
And maybe make the `handler` an actual handler